### ISSUE TITLE
Automatically choose less-used GPU

### ISSUE THREAD

**Issue by hbredin** [pyannote/pyannote-audio#173]
One solution is to write a wrapper around `nvidia-smi` but this is not pretty.

Here is an example I found: 

https://github.com/open-mmlab/mmdetection/blob/65bef5e29ebe149547f077eee45e6ad9ba8ae94f/mmdet/core/utils/misc.py#L40-L53

**Comment by mogwai** [comment]
I think #407 might help deal with this?

**Comment by hbredin** [comment]
Yes, this will most likely be handled automagically by pytorch-lightning. Linking the PR.
